usaspending_api/disaster/v2/views/disaster_base.py

```diff
@@ -1,11 +1,86 @@
 from django.utils.functional import cached_property
 from rest_framework.views import APIView
+from datetime import datetime, timezone
 
+from django.db.models import Exists, OuterRef, Max
 from usaspending_api.awards.v2.lookups.lookups import award_type_mapping
 from usaspending_api.common.data_classes import Pagination
 from usaspending_api.common.validator import customize_pagination_with_sort_columns, TinyShield
 from usaspending_api.references.models import DisasterEmergencyFundCode
+from usaspending_api.submissions.models import DABSSubmissionWindowSchedule
+from usaspending_api.references.models.gtas_sf133_balances import GTASSF133Balances
+from usaspending_api.awards.models.financial_accounts_by_awards import FinancialAccountsByAwards
 from usaspending_api.submissions.helpers import get_last_closed_submission_date
+from usaspending_api.submissions.models import SubmissionAttributes
+from django.db.models.functions import Concat
+from django.db.models import Value, CharField
+
+COVID_19_GROUP_NAME = "covid_19"
+
+
+def covid_def_codes():
+    return DisasterEmergencyFundCode.objects.filter(group_name=COVID_19_GROUP_NAME)
+
+
+def covid_def_code_strings():
+    return [code["code"] for code in covid_def_codes().values("code")]
+
+
+def latest_gtas_of_each_year_queryset():
+    return (
+        GTASSF133Balances.objects.annotate(
+            is_highest_period=Exists(
+                GTASSF133Balances.objects.values("fiscal_year")
+                .annotate(
+                    fiscal_period_max=Max("fiscal_period"),
+                    include=Exists(
+                        DABSSubmissionWindowSchedule.objects.filter(
+                            is_quarter=False,
+                            submission_fiscal_year=OuterRef("fiscal_year"),
+                            submission_fiscal_month=OuterRef("fiscal_period"),
+                            submission_reveal_date__lte=datetime.now(timezone.utc),
+                        )
+                    ),
+                )
+                .filter(
+                    include=True,
+                    fiscal_year=OuterRef("fiscal_year"),
+                    fiscal_period_max=OuterRef("fiscal_period"),
+                    disaster_emergency_fund_code__in=covid_def_code_strings(),
+                )
+                .values("fiscal_year", "fiscal_period_max")
+            )
+        )
+        .annotate(fiscal_period_and_year=Concat("fiscal_year", Value("/"), "fiscal_period", output_field=CharField()))
+        .filter(
+            is_highest_period=True,
+            fiscal_period_and_year__in=[
+                f"{elem['submission_fiscal_year']}/{elem['submission_fiscal_month']}"
+                for elem in DABSSubmissionWindowSchedule.objects.filter(is_quarter=False)
```

Comment:
I'll have to see the SQL on this, but `DABSSubmissionWindowSchedule` was already interrogated above. Not sure if it needs to be brought into this query again.